New release 2.10 for Unicode 16 support

I'd like to have / create a new release 2.10 that supports Unicode 16. I'd be happy to create the release if others give the go-ahead.

(January 8 will be the feature freeze for Julia 1.12; if possible I'd like to have Unicode 16 in Julia 1.12.)

I'm getting some warnings that it would be good to fix:
```
utf8proc.c:702:28: warning: implicit conversion changes signedness: 'utf8proc_uint32_t' (aka 'unsigned int') to 'utf8proc_int32_t' (aka 'int') [-Wsign-conversion]
  702 |                 *starter = composition;
      |                          ~ ^~~~~~~~~~~
utf8proc.c:691:28: warning: comparison of integers of different signs: 'utf8proc_int32_t' (aka 'int') and 'utf8proc_uint32_t' (aka 'unsigned int') [-Wsign-compare]
  691 |           if (current_char <= max_second) {
      |               ~~~~~~~~~~~~ ^  ~~~~~~~~~~
utf8proc.c:695:32: warning: comparison of integers of different signs: 'utf8proc_int32_t' (aka 'int') and 'utf8proc_uint32_t' (aka 'unsigned int') [-Wsign-compare]
  695 |               if (current_char < second) {
      |                   ~~~~~~~~~~~~ ^ ~~~~~~
utf8proc.c:699:32: warning: comparison of integers of different signs: 'utf8proc_int32_t' (aka 'int') and 'utf8proc_uint32_t' (aka 'unsigned int') [-Wsign-compare]
  699 |               if (current_char == second) {
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,7 @@ include (utils.cmake)
 
 disallow_intree_builds()
 
+# API version - be sure to update utf8proc.h and Makefile, too!
 project (utf8proc VERSION 2.10.0 LANGUAGES C)
 
 # This is the ABI version number, which may differ from the

--- a/MANIFEST
+++ b/MANIFEST
@@ -2,8 +2,8 @@ include/
 include/utf8proc.h
 lib/
 lib/libutf8proc.a
-lib/libutf8proc.so -> libutf8proc.so.3.0.0
-lib/libutf8proc.so.2 -> libutf8proc.so.3.0.0
-lib/libutf8proc.so.3.0.0
+lib/libutf8proc.so -> libutf8proc.so.3.1.0
+lib/libutf8proc.so.2 -> libutf8proc.so.3.1.0
+lib/libutf8proc.so.3.1.0
 lib/pkgconfig/
 lib/pkgconfig/libutf8proc.pc

--- a/Makefile
+++ b/Makefile
@@ -23,11 +23,11 @@ SOFLAG = -Wl,-soname
 # The API version number is defined in utf8proc.h.
 # Be sure to also update these ABI versions in MANIFEST and CMakeLists.txt!
 MAJOR=3
-MINOR=0
+MINOR=1
 PATCH=0
 
 # api version (also in utf8proc.h and CMakeLists.txt)
-VERSION=2.9.0
+VERSION=2.10.0
 
 OS := $(shell uname)
 ifeq ($(OS),Darwin) # MacOS X

--- a/NEWS.md
+++ b/NEWS.md
@@ -452,3 +452,5 @@ Release of version 1.0.1
 [#233]: https://github.com/JuliaStrings/utf8proc/issues/233
 [#247]: https://github.com/JuliaStrings/utf8proc/issues/247
 [#253]: https://github.com/JuliaStrings/utf8proc/issues/253
+[#270]: https://github.com/JuliaStrings/utf8proc/issues/270
+[#277]: https://github.com/JuliaStrings/utf8proc/issues/277

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,5 +1,14 @@
 # utf8proc release history #
 
+## Version 2.10.0 ##
+
+2024-12-31
+
+ - Unicode 16 support ([#277]).
+ - New `utf8proc_charwidth_ambiguous` function to return whether a character has
+   East Asian width class A (Ambiguous) ([#270]).
+
+
 ## Version 2.9.0 ##
 
 2023-10-20

--- a/utf8proc.h
+++ b/utf8proc.h
@@ -71,7 +71,7 @@
 /** The MAJOR version number (increased when backwards API compatibility is broken). */
 #define UTF8PROC_VERSION_MAJOR 2
 /** The MINOR version number (increased when new functionality is added in a backwards-compatible manner). */
-#define UTF8PROC_VERSION_MINOR 9
+#define UTF8PROC_VERSION_MINOR 10
 /** The PATCH version (increased for fixes that do not change the API). */
 #define UTF8PROC_VERSION_PATCH 0
 /** @} */

